### PR TITLE
feat: operator set release

### PR DESCRIPTION
**Pull request by stevennevins**

### CODE REVIEW

File: src/ServiceManagerBase.sol [hunk 1]
```diff
@@ -135,6 +140,77 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
         _setRewardsInitiator(newRewardsInitiator);
     }
 
+    function migrateAndCreateOperatorSetIds(uint32[] memory operatorSetsToCreate) external onlyOwner{
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol]: comments on all funcs

File: src/ServiceManagerBase.sol [hunk 2]
```diff
@@ -135,6 +140,77 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
...
+        migrationFinalized = true;
+    }
+
+    function _migrateAndCreateOperatorSetIds(uint32[] memory operatorSetIdsToCreate) internal {
+        _avsDirectory.becomeOperatorSetAVS();
+        IAVSDirectory(address(_avsDirectory)).createOperatorSets(operatorSetIdsToCreate);
+    }
+
+    function _migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) internal {
+        IAVSDirectory(address(_avsDirectory)).migrateOperatorsToOperatorSets(operators, operatorSetIds);
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol]: are we not checking that the provided operators are correct?

Comment (gpsanant) [reply src/ServiceManagerBase.sol]: there's a race condition here where operators deregister for a certain quorum inbetween when the view function is called and migrate is called

Comment (ypatil12) [reply src/ServiceManagerBase.sol]: We can and just skip. May be cleaner

Comment (ypatil12) [reply src/ServiceManagerBase.sol]: Regarding race condition, we can also prevent m2 deregistrations in AVSDirectory

Comment (stevennevins) [reply src/ServiceManagerBase.sol]: yeah we should block m2 functions in core  
`if (isOperatorSetAVS[avs]==true)`

File: src/ServiceManagerBase.sol (hunk 1 above)
Comment (stevennevins) [reply src/ServiceManagerBase.sol]: [12eda5d](https://github.com/Layr-Labs/eigenlayer-middleware/pull/288/commits/12eda5dd36ba793b03136516d2342554b173151a)

File: src/ServiceManagerBase.sol (hunk 2 above)
Comment (stevennevins) [reply src/ServiceManagerBase.sol]: added additional checks [d64f4af](https://github.com/Layr-Labs/eigenlayer-middleware/pull/288/commits/d64f4af465201d66e2c581e42f28b32764a82381)

Comment (ypatil12) [reply src/ServiceManagerBase.sol]: added pr here to block m2 dereg: https://github.com/Layr-Labs/eigenlayer-contracts/pull/688

**Comment by stevennevins** [comment]
Resolved the code size issue by wrapping the library from core

**Pull request closed without merging**